### ISSUE TITLE
GOPATH does not take effect

### ISSUE THREAD

**Issue by shellus** [golang/go#20099]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
go version go1.8.1 windows/amd64

### What operating system and processor architecture are you using (`go env`)?
set GOARCH=amd64
set GOBIN=
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\Users\shellus\go
set GORACE=
set GOROOT=C:\Go
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\shellus\AppData\Local\Temp\go-build143925707=/tmp/go-build -gno-record-gcc-switches
set CXX=g++
set CGO_ENABLED=1
set PKG_CONFIG=pkg-config
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2

### What did you do?

If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
A link on play.golang.org is best.

![image](https://cloud.githubusercontent.com/assets/3044880/25339050/f6a4da8a-2933-11e7-88bb-a572e82ca925.png)
GOROOT is C:\Go  GOPATH is not seting (I think it will use the default {user}/go)

```
C:\Users\shellus\go>tree
├─bin
├─pkg
│  └─windows_amd64
│      └─github.com
│          ├─akavel
│          │  └─rsrc
│          ├─AllenDang
│          ├─Knetic
│          └─lxn
│              └─walk
└─src
    └─github.com
        ├─akavel
        │  └─rsrc
        │      ├─binutil
        │      ├─coff
        │      └─ico
        ├─AllenDang
        │  └─w32
        ├─Knetic
        │  └─govaluate
        └─lxn
            ├─walk
            │  ├─data
            │  ├─declarative
            │  ├─examples
            │  │  ├─actions
            │  │  ├─clipboard
            │  │  ├─databinding
            │  │  ├─drawing
            │  │  ├─dropfiles
            │  │  ├─externalwidgets
            │  │  ├─filebrowser
            │  │  ├─gradientcomposite
            │  │  ├─imageicon
            │  │  ├─imageviewer
            │  │  ├─img
            │  │  ├─listbox
            │  │  ├─logview
            │  │  ├─multiplepages
            │  │  ├─notifyicon
            │  │  ├─progressindicator
            │  │  ├─radiobutton
            │  │  ├─settings
            │  │  ├─slider
            │  │  ├─statusbar
            │  │  ├─tableview
            │  │  └─webview
            │  ├─l10n
            │  └─tools
            │      └─ui2walk
            └─win

C:\Users\shellus\go>

```

i see GOPATH is {user}/go

but i usage makefile content:
```
export GOPATH:=$(shell pwd)

default: window

window:
	go install -tags 'release' win32api/createWindow
```

```
C:\data\c\my-go-starter>tree
C:.
├─.idea
│  └─libraries
├─bin
└─src
    ├─base
    ├─echo
    ├─gui
    ├─http
    ├─parse
    ├─signal
    ├─socket
    ├─syscall
    └─win32api
        ├─createWindow
        └─messageBox


C:\data\c\my-go-starter>make window
go install -tags 'release' win32api/createWindow
src\win32api\createWindow\createWindow.go:4:2: cannot find package "github.com/AllenDang/w32" in any of:
        C:\Go\src\github.com\AllenDang\w32 (from $GOROOT)
        c:\data\c\my-go-starter\src\github.com\AllenDang\w32 (from $GOPATH)
make: *** [window] Error 1

C:\data\c\my-go-starter>

```

How can i set GOPATH it up?

**Comment by matipan** [comment]
Hi!
So, the problem here is that you are not using the `GOPATH` correctly.
If you read(and you should) [this section](https://golang.org/doc/code.html#GOPATH) of How to Write Go Code, you'll understand that there can only be one `GOPATH` set, meaning that you can't have per-project GOPATHs. 

In your example, you've used the default GOPATH(which is {user}/go) to `go get` a bunch of packages, then you created a program outside of the GOPATH, tried to set GOPATH to a new location and run go install. The problem here is that all the packages you need for your program are inside the previous GOPATH, which is why the `go install` tool is telling you that it cannot find them.

To make this work you need to remove the `export GOPATH:=$(shell pwd)` line from your Makefile, since I already said that you can't have per-project GOPATHs, and set your GOPATH again to be `C:\Users\shellus\go>`.

In conclusion, the GOPATH is working as intended and it is in fact taking effect

**Comment by shellus** [comment]
@matipan Can GOPATH not use multiple directories?

**Comment by matipan** [comment]
If you want, you can have multiple workspaces. But take into account that the go tool will look for the packages your program imports inside `GOPATH/src`, so if you change the GOPATH to a different workspace that doesn't have the packages you need, the go tool will fail to build it.

To understand how the GOPATH works, you should read [this section](https://golang.org/doc/code.html#GOPATH).
For more detailed information and examples see 'go help gopath'.

**Comment by mvdan** [comment]
For questions about Go, see https://github.com/golang/go/wiki/Questions.
